### ISSUE TITLE
add control of versions of external libraries in ./configure

### ISSUE THREAD

**Issue by dimpase** [Macaulay2/M2#1024]
It is not so hard (I can do this if we agree to go forward with this) to add checks for versions of external libraries in ./configure to match ones in `libraries/*/Makefile.in`.

One further question here is whether we want to be a bit more fine-grained and have optional
`MINVERSION = <blah>` in these files, so that the test would check that the version of the library is at least `MINVERSION`, if it is set, and otherwise use `VERSION`. 

(I've been bitten by this a number of times, I guess I'm not alone, so this would be a reasonable improvement, IMHO)

**Comment by dimpase** [comment]
@antonleykin

**Comment by DanGrayson** [comment]
I don't see the point.  I prefer to install a version test only if we have some concrete reason -- where we know that an earlier version couldn't possibly work.  Better than a version test would be a feature test, if that's the point.

**Comment by DanGrayson** [comment]
In any case, there is no reason to try to match the version numbers in our library makefiles, as those are intended to always be the very latest version.

**Comment by dimpase** [comment]
A feature test is often just not possible, as the problem can manifest itself only on some platforms (and having different library versions on different platforms is hard), or only in a hard to reproduce way. A case at hand is https://github.com/antonleykin/M2/pull/5

**Comment by dimpase** [comment]
How do you test for "random crashes", or "much improved performance", or "better documentation"? 
There are plenty of cases for just doing version tests. I'm not making this up, I'm telling from the experience of unvendoring SageMath dependencies, many many more than in M2, see
https://trac.sagemath.org/ticket/27330

**Comment by mahrud** [comment]
@dimpase the CMake build accomplishes this in the `cmake/check-libraries.cmake` file. Would you like to keep this open for the same feature using autotools, or should it be closed?

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Closing this, because `./configure` already does what is being asked — and the one part it
deliberately does not do is the part @DanGrayson objected to above.

**Version checks are there, and some predate this issue.** `M2/configure.ac` carries minimum-version
tests for flint (`>= 2.6.3`), mpfr (`>= 4`, needed for `mpfr_nrandom`), tbb (`>= 2020`), boost
(`>= 1.65`), readline (`>= 6`), gmp (`>= 6.2.1`) and factory. The factory one is spelled with exactly
the `MINVERSION` variable proposed here:

```sh
FACTORY_MINVERSION=4.2.0
if $PKG_CONFIG --exists "factory >= $FACTORY_MINVERSION"
```

That line was added in `8b2e430b46`, in 2015 — four years before this issue. Feature tests are used
too where they are sharper, which is @DanGrayson's stated preference:
`AC_CHECK_HEADERS([flint/nmod.h])`, commented *"only present in flint >= 2.9.0"*.

**What is deliberately not done is matching `M2/libraries/*/Makefile.in`.** The two serve different
purposes: `configure` tests a *system* library to decide whether it will work with M2, while
`M2/libraries` builds a library for us when the system's will not do. So the numbers are meant to
differ, and they do:

| | `configure` accepts | `M2/libraries` builds |
| --- | --- | --- |
| flint | `>= 2.6.3` | 3.6.0 |
| factory | `>= 4.2.0` | 4.4.1 |
| mpfr | `>= 4` | 4.2.2 |

That gap is the point. `M2/libraries` tracks the newest release; `configure` accepts anything known
to work, which is what lets M2 build against libraries already present on older systems. Tightening
the first to match the second would remove that, which is what @DanGrayson meant by *"there is no
reason to try to match the version numbers in our library makefiles."*

@mahrud — to answer your 2020 question directly: closing, rather than keeping it open for an
autotools equivalent of `cmake/check-libraries.cmake`. Autotools has its own checks; they are just
not written as a version table.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>

**Comment by dimpase** [comment]
I would be surprised if using flint2 is still feasible - in particular as there are no CI tests checking this.

This probably is applicable to a number of other dependencies with loose version bounds (or other version-constraining tests)

**Comment by d-torrance** [comment]
I have an Ubuntu PPA that does daily builds of the development branch on releases going back to 18.04, so we're definitely checking that some of these older libraries still work.  The 18.04 package in particular builds against FLINT 2.6.3.

**Comment by dimpase** [comment]
well, would flint2 work with the current branch? 

old versions of M2 are a different story

**Comment by d-torrance** [comment]
Yup!  Here's a build log from a couple weeks ago (after #4434 was merged, triggering a new PPA build) in Ubuntu 18.04 using FLINT 2.6.3:

https://launchpadlibrarian.net/872302907/buildlog_ubuntu-bionic-amd64.macaulay2_1.26.06+git202608042342-0ppa202607182143~ubuntu18.04.1_BUILDING.txt.gz
